build/release/release.sh

```diff
@@ -16,6 +16,35 @@ if [[ $1x = x ]]; then
     TAG=$DATE-$VERSION
 else
     TAG=$1
+
```

Comment:
while we are here, should we remove the date-version fallback above? there's no way for it to pass the new validation, since the output isn't in the `X.X.X` semver format produced by `lnd --version`

Reply:
instead we should just fail to build if it no `TAG` is specified

Reply:
I don't know what the original intention of the fallback is. But I can imagine for local testing of the script?

For the date-version fallback, we don't validate so the script should still work.

Reply:
The date fallback was just an old default, but with this new verification logic it can be used to just make sure things build without needing to checkout out a tag, etc.